scripts/react-compiler-compliance-check.ts

```diff
@@ -261,15 +262,163 @@ function createFilesGlob(filesToCheck?: string[]): string | undefined {
+    const changedFiles = new Set(diffResult.files.map((file) => file.filePath));
+    const filteredSuccesses = new Set<string>();
+    results.success.forEach((file) => {
+        if (!changedFiles.has(file)) {
+            return;
+        }
+        filteredSuccesses.add(file);
+    });
+
+    if (shouldPrintSuccesses) {
+        if (filteredSuccesses.size === 0) {
+            logInfo('No successes remain after filtering by diff.');
+        } else {
+            logInfo(`${filteredSuccesses.size} out of ${results.success.size} successes remain after filtering by diff.`);
+        }
+    }
+
+    if (shouldPrintSuppressedErrors) {
+        if (filteredSuppressedFailures.size === 0) {
+            logInfo('No suppressed errors remain after filtering by diff.');
+        } else {
+            logInfo(`${filteredSuppressedFailures.size} out of ${results.suppressedFailures.size} successes remain after filtering by diff.`);
+        }
+    }
+
+    if (filteredFailures.size === 0) {
+        logInfo('No failures remain after filtering by diff.');
+    } else {
+        logInfo(`${filteredFailures.size} out of ${results.failures.size} failures remain after filtering by diff.`);
+    }
+
+    return {
+        success: filteredSuccesses,
+        failures: filteredFailures,
+        suppressedFailures: results.suppressedFailures,
+    };
+}
+
+function printResults({success, failures, suppressedFailures}: CompilerResults, {shouldPrintSuccesses, shouldPrintSuppressedErrors}: PrintResultsOptions): void {
+    if (shouldPrintSuccesses && success.size > 0) {
         log();
         logSuccess(`Successfully compiled ${success.size} files with React Compiler:`);
+        log();
+
+        success.forEach((successFile) => {
+            logSuccess(`${successFile}`);
+        });
+
+        log();
+    }
+
+    if (shouldPrintSuppressedErrors && suppressedFailures.size > 0) {
+        // Create a Map of suppressed error type -> Failure[] with distinct errors and a list of failures with that error
+        const suppressedErrorMap = new Map<string, CompilerFailure[]>();
+        suppressedFailures.forEach((failure) => {
+            if (!failure.reason) {
+                return;
+            }
+
+            if (!suppressedErrorMap.has(failure.reason)) {
+                suppressedErrorMap.set(failure.reason, []);
+            }
+
+            suppressedErrorMap.get(failure.reason)?.push(failure);
+        });
+
+        log();
+        logWarn(`Suppressed the following errors in these files:`);
+        log();
+
+        for (const [error, suppressedErrorFiles] of suppressedErrorMap.entries()) {
+            logBold(error);
+            const filesLine = suppressedErrorFiles.map((failure) => getUniqueFileKey(failure)).join(', ');
+            logNote(`${TAB} - ${filesLine}`);
+        }
+
+        log();
     }
 
-    const tab = '    ';
+    const isPassed = failures.size === 0;
+    if (isPassed) {
```

Comment:
NAB remove redundant local variable:


```suggestion
    if (failures.size === 0) {
```

Reply:
Personally, i always prefer to have local named variables like this to make the logic more explicit.

To me, `isPassed` tells more clearly what this condition is about than `failures.size === 0`.

I can change this ofc

Reply:
NAB - it's totally personal preference